// @voiceflow/general-types/build/version/settings.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultGeneralVersionSettings = exports.defaultBaseVersionSettings = exports.defaultPrompt = exports.SessionType = exports.RepeatType = void 0;
const types_1 = require("../types");
var RepeatType;
(function (RepeatType) {
    RepeatType["OFF"] = "OFF";
    RepeatType["ALL"] = "ALL";
    RepeatType["DIALOG"] = "DIALOG";
})(RepeatType = exports.RepeatType || (exports.RepeatType = {}));
var SessionType;
(function (SessionType) {
    SessionType["RESUME"] = "resume";
    SessionType["RESTART"] = "restart";
})(SessionType = exports.SessionType || (exports.SessionType = {}));
const defaultPrompt = (prompt, defaultVoice) => {
    if (!(prompt === null || prompt === void 0 ? void 0 : prompt.content)) {
        return null;
    }
    return {
        voice: prompt.voice || defaultVoice,
        content: prompt.content,
    };
};
exports.defaultPrompt = defaultPrompt;
const defaultBaseVersionSettings = ({ error, repeat = RepeatType.ALL, session = { type: SessionType.RESTART }, defaultVoice = null, defaultCanvasNodeVisibility = null, } = {}, { defaultPromptVoice }) => ({
    error: exports.defaultPrompt(error, defaultVoice !== null && defaultVoice !== void 0 ? defaultVoice : defaultPromptVoice),
    repeat,
    session,
    defaultVoice,
    defaultCanvasNodeVisibility,
});
exports.defaultBaseVersionSettings = defaultBaseVersionSettings;
const defaultGeneralVersionSettings = (_a = {}) => {
    var { locales = [types_1.Locale.EN_US] } = _a, baseSettings = __rest(_a, ["locales"]);
    return (Object.assign(Object.assign({}, exports.defaultBaseVersionSettings(baseSettings, { defaultPromptVoice: types_1.Voice.DEFAULT })), { locales }));
};
exports.defaultGeneralVersionSettings = defaultGeneralVersionSettings;
