// @voiceflow/general-types/build/nodes/types.js
"use strict";
/* eslint-disable camelcase */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommandType = exports.EventType = exports.ConditionsLogicInterface = exports.IntegrationPlatform = exports.IntegrationType = exports.NodeType = exports.TraceType = void 0;
var TraceType;
(function (TraceType) {
    TraceType["END"] = "end";
    TraceType["PATH"] = "path";
    TraceType["FLOW"] = "flow";
    TraceType["SPEAK"] = "speak";
    TraceType["BLOCK"] = "block";
    TraceType["DEBUG"] = "debug";
    TraceType["CHOICE"] = "choice";
    TraceType["STREAM"] = "stream";
    TraceType["VISUAL"] = "visual";
})(TraceType = exports.TraceType || (exports.TraceType = {}));
var NodeType;
(function (NodeType) {
    NodeType["SPEAK"] = "speak";
    NodeType["START"] = "start";
    NodeType["CARD"] = "card";
    NodeType["INTERACTION"] = "interaction";
    // logic
    NodeType["SET"] = "set";
    NodeType["SET_V2"] = "setV2";
    NodeType["IF"] = "if";
    NodeType["IF_V2"] = "ifV2";
    NodeType["RANDOM"] = "random";
    NodeType["CAPTURE"] = "capture";
    // integrations
    NodeType["API"] = "api";
    NodeType["ZAPIER"] = "zapier";
    NodeType["INTEGRATIONS"] = "integrations";
    NodeType["GOOGLE_SHEETS"] = "google_sheets";
    // advanced
    NodeType["INTENT"] = "intent";
    NodeType["FLOW"] = "flow";
    NodeType["CODE"] = "code";
    NodeType["EXIT"] = "exit";
    NodeType["PROMPT"] = "prompt";
    NodeType["COMMAND"] = "command";
    NodeType["TRACE"] = "trace";
    // other
    NodeType["STREAM"] = "stream";
    NodeType["VISUAL"] = "visual";
    NodeType["GENERAL"] = "general";
    NodeType["DIRECTIVE"] = "directive";
    NodeType["DEPRECATED"] = "deprecated";
})(NodeType = exports.NodeType || (exports.NodeType = {}));
// Integrations Step
var IntegrationType;
(function (IntegrationType) {
    IntegrationType["ZAPIER"] = "Zapier";
    IntegrationType["CUSTOM_API"] = "Custom API";
    IntegrationType["GOOGLE_SHEETS"] = "Google Sheets";
})(IntegrationType = exports.IntegrationType || (exports.IntegrationType = {}));
var IntegrationPlatform;
(function (IntegrationPlatform) {
    IntegrationPlatform["ZAPIER"] = "Zapier";
    IntegrationPlatform["GOOGLE_SHEETS"] = "Google Sheets";
})(IntegrationPlatform = exports.IntegrationPlatform || (exports.IntegrationPlatform = {}));
// New
var ConditionsLogicInterface;
(function (ConditionsLogicInterface) {
    ConditionsLogicInterface["VARIABLE"] = "variable";
    ConditionsLogicInterface["VALUE"] = "value";
    ConditionsLogicInterface["LOGIC_GROUP"] = "logic_group";
    ConditionsLogicInterface["EXPRESSION"] = "expression";
})(ConditionsLogicInterface = exports.ConditionsLogicInterface || (exports.ConditionsLogicInterface = {}));
// BUILT IN EVENTS
var EventType;
(function (EventType) {
    EventType["INTENT"] = "intent";
})(EventType = exports.EventType || (exports.EventType = {}));
var CommandType;
(function (CommandType) {
    CommandType["JUMP"] = "jump";
    CommandType["PUSH"] = "push";
})(CommandType = exports.CommandType || (exports.CommandType = {}));
