// @voiceflow/eslint-config/config/react.js
module.exports = {
  extends: ['plugin:react/recommended', 'plugin:jsx-a11y/recommended'],
  plugins: ['react', 'react-hooks', 'jsx-a11y'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  rules: {
    // error
    'react/jsx-fragments': 'error',

    'react-hooks/rules-of-hooks': 'error',

    // warn
    'react/jsx-curly-brace-presence': 'warn',
    'react/jsx-filename-extension': ['warn', { extensions: ['.jsx', '.tsx'] }],

    'jsx-a11y/click-events-have-key-events': 'warn',
    'jsx-a11y/label-has-for': 'warn',
    'jsx-a11y/label-has-associated-control': 'warn',
    'jsx-a11y/no-static-element-interactions': 'warn',

    // off
    'react/destructuring-assignment': 'off',
    'react/no-unescaped-entities': 'off',
    'react/prop-types': 'off',

    // high performance overhead & low value
    'react/display-name': 'off',
    'react/no-string-refs': 'off',
    'react/require-render-return': 'off',
    'react/no-direct-mutation-state': 'off',
    'react/no-deprecated': 'off',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
};
