// @voiceflow/general-types/build/constants/device.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEVICE_SIZE_MAP = exports.DeviceType = void 0;
var DeviceType;
(function (DeviceType) {
    DeviceType["MOBILE"] = "mobile";
    DeviceType["TABLET"] = "tablet";
    DeviceType["DESKTOP"] = "desktop";
    DeviceType["SMART_WATCH"] = "smart_watch";
    DeviceType["TELEVISION"] = "television";
    DeviceType["IN_CAR_DISPLAY"] = "in_car_display";
    DeviceType["ECHO_SPOT"] = "echo_spot";
    DeviceType["ECHO_SHOW_8"] = "echo_show_8";
    DeviceType["ECHO_SHOW_10"] = "echo_show_10";
    DeviceType["FIRE_HD_8"] = "fire_hd_8";
    DeviceType["FIRE_HD_10"] = "fire_hd_10";
    DeviceType["FIRE_TV_CUBE"] = "fire_tv_cube";
    DeviceType["GOOGLE_NEST_HUB"] = "google_nest_hub";
})(DeviceType = exports.DeviceType || (exports.DeviceType = {}));
exports.DEVICE_SIZE_MAP = {
    [DeviceType.MOBILE]: { width: 375, height: 812 },
    [DeviceType.TABLET]: { width: 1024, height: 1366 },
    [DeviceType.DESKTOP]: { width: 1440, height: 900 },
    [DeviceType.SMART_WATCH]: { width: 184, height: 224 },
    [DeviceType.TELEVISION]: { width: 1920, height: 1200 },
    [DeviceType.IN_CAR_DISPLAY]: { width: 1200, height: 800 },
    [DeviceType.ECHO_SPOT]: { width: 480, height: 480 },
    [DeviceType.ECHO_SHOW_8]: { width: 1280, height: 800 },
    [DeviceType.ECHO_SHOW_10]: { width: 1280, height: 800 },
    [DeviceType.FIRE_HD_8]: { width: 1280, height: 800 },
    [DeviceType.FIRE_HD_10]: { width: 1920, height: 1200 },
    [DeviceType.FIRE_TV_CUBE]: { width: 1920, height: 1080 },
    [DeviceType.GOOGLE_NEST_HUB]: { width: 1280, height: 730 },
};
